File: api/python/examples/rpc/rpc_service.py
#encoding=utf8

from zbus import Service,ServiceConfig, RpcServiceHandler, Remote, SingleBroker

class MyService(object): 
    @Remote()
    def getString(self, ping):
        return ping
    
    @Remote()
    def echo(self, ping):
        return ping
    
    @Remote()
    def save(self, user):
        print user
        return 'OK'
        
    @Remote('user')
    def user(self, username):
        return {'Name': username, 'Addr': u'中文'}
    
    @Remote()
    def plus(self, a, b):
        print 'plus(%s,%s)'%(a, b)
        return a + b 
 
handler = RpcServiceHandler()
handler.add_module('Interface', MyService())


config = ServiceConfig()
broker = SingleBroker(host = 'localhost', port = 15555)
config.broker = broker 
config.service_name = 'MyRpc'
config.service_andler = handler
config.thread_count = 20

svc = Service(config)
svc.start()
svc.join()

File: api/csharp/Examples/rpc/RpcService.cs
using System;  

using Zbus.Broker;
using Zbus.RPC;

namespace Zbus.Examples
{
    class User
    {
        public string name;
        public string addr;
    }

    class MyService
    { 
        public string getString(string msg)
        {
            return msg;
        }
         
        public int plus(int a, int b)
        {
            return a + b;
        }
         
        public User user(string name)
        {
            User user = new User();
            user.name = name;
            user.addr = "深圳";
            return user; 
        }

        [Remote(Exclude=true)]
        public void ignore()
        {

        }
    }

    class MyService2
    { 
        public string getString(string msg)
        {
            return msg;
        } 
    }
    
    
    class RpcServiceTest
    {
        public static void Main(string[] args)
        { 
            IBroker broker = new SingleBroker(); //using default configuration

            RpcProcessor processor = new RpcProcessor();
            processor.AddModule(new MyService());
            processor.AddModule(new MyService2());

            ServiceConfig config = new ServiceConfig(broker);
            config.Mq = "MyRpc"; //Service entry in zbus as a MQ
            config.MessageProcessor = processor;
            config.ConsumerCount = 32;
 
            Service service = new Service(config);
            service.Start();
            //service.Stop();
            
            //broker.Dispose(); 
            
            //Console.ReadKey();
        } 
    }
}
